### PR TITLE
Switch to upstream `go-ora`

### PR DESCRIPTION
**Pull request by Tener**
Updates https://github.com/gravitational/teleport/issues/43544.

### CODE REVIEW

**Review by tigrato — COMMENTED** [review]
should we wait for the proper v2.8.22 release instead of pinning a random master commit?

**Comment by Tener** [comment]
> should we wait for the proper v2.8.22 release instead of pinning a random master commit?

That is an option, however: 
- I don't know when a next release may go out that includes https://github.com/sijms/go-ora/pull/608. 
- I've checked all following commits and those look good, so it is fine to pull a tip of master instead of just https://github.com/sijms/go-ora/commit/eaedcffe942949d13be8e1fbb8dc695e8938c40d
- We have very little code that actually depends on this library and I have manually verified it works after the upgrade.

Let me know what you think.

**Comment by greedy52** [comment]
> Let me know what you think.

Agreed. Anything is better than what's currently in `go.mod`. Release testing likely starts early next week. If no release from upstream this week, I personally would prefer this by then.

Marking as part of #43544

**Review by greedy52 — APPROVED** [review]

**Review by tigrato — APPROVED** [review]
Once you update the go.mod

File: integrations/event-handler/go.mod [hunk 1]
```diff
@@ -343,7 +343,7 @@ replace (
 	// replace module github.com/moby/spdystream until https://github.com/moby/spdystream/pull/91 merges and deps are updated
 	// otherwise tests fail with a data race detection.
 	github.com/moby/spdystream => github.com/gravitational/spdystream v0.0.0-20230512133543-4e46862ca9bf
-	github.com/redis/go-redis/v9 => github.com/gravitational/redis/v9 v9.0.2-teleport.2
+	github.com/redis/go-redis/v9 => github.com/gravitational/redis/v9 v9.0.2-teleport.3
```
Comment (greedy52) [inline integrations/event-handler/go.mod:346]:
```suggestion
	github.com/redis/go-redis/v9 => github.com/gravitational/redis/v9 v9.0.2-teleport.2
```
Revert this?

**Pull request merged**